compiler/modules/types/list.bal

```diff
@@ -32,7 +42,7 @@ public class ListDefinition {
         }
     }
 
-    public function define(Env env, SemType[] members, SemType rest) returns ComplexSemType {
+    public function define(Env env, FixedLengthArray members, SemType rest) returns ComplexSemType {
         ListAtomicType rwType = { members: members.cloneReadOnly(), rest };
```

Comment:
Somewhere we need to deal with normalizing the FixedLengthArray by chopping duplicates off the end of the `initial` array.  Feels like this should be done here with a fixedLengthNormalize function.